core/api.py

```diff
@@ -119,3 +137,76 @@ def by_date(self, date:str = 'today', period: str = '1d' ) -> dict:
+        start_date: str = 'today',
+        end_date: str = 'today',
+        detail_level: str = '1min',
+        start_time: str = None,
+        end_time: str = None,) -> dict:
+        """Heart rate intraday by interval.
+
+        Time parameters are optional.
+
+        API endpoint returns inconsistent results depending on the parameters.
+        The following table documents the results of different parameter combinations:
+
+        start_date   | end_date     | detail_level | start_time | end_time | result
+        ---------------------------------------------------------------------
+        'today'      | 'today'      | '1min'       | None       | None     | daily + intraday
+        'today'      | 'today'      | '1sec'       | None       | None     | invalid 
+        '2022-08-14' | '2022-08-14' | '1min        | None       | None     | daily + intraday
+        '2022-08-14' | '2022-08-14' | '1sec        | None       | None     | generic activity categories
+        '2022-08-11' | '2022-08-14' | '1min"       | None       | None     | daily only
+        '2022-08-11' | '2022-08-14' | '1sec"       | None       | None     | generic activity categories
+        'today'      | 'today'      | '1min'       | '07:00'    | '13:00'  | daily + timeboxed intraday
+        'today'      | 'today'      | '1sec'       | '07:00'    | '13:00'  | daily + timeboxed intraday at 1 sec intervals
+        '2022-08-14' | '2022-08-14' | '1min        | '07:00'    | '13:00'  | daily + timeboxed intraday
+        '2022-08-14' | '2022-08-14' | '1sec'       | '07:00'    | '13:00'  | daily + timeboxed intraday at 1 sec intervals
+        '2022-08-11' | '2022-08-14' | '1min        | '07:00'    | '13:00'  | invalid
+        '2022-08-11' | '2022-08-14' | '1sec'       | '07:00'    | '13:00'  | invalid
+
+        Reference:
+        https://dev.fitbit.com/build/reference/web-api/heartrate-timeseries/get-heartrate-timeseries-by-date/
+
+        Args:
+            start_date (str, optional): yyyy-MM-dd format date, or 'today'.. Defaults to 'today'.
+            end_date (str, optional): yyyy-MM-dd format date, or 'today'.. Defaults to 'today'.
+            detail_level (str, optional): '1sec' or '1min'. Defaults to '1min'.
+            start_time (str, optional): 'HH:mm' format. Defaults to None.
+            end_time (str, optional): 'HH:mm' format. Defaults to None.
+
+        Raises:
+            ValueError: For unsupported parameters, or if only one of start/end times provided.
+
+        Returns:
+            dict: from JSON data
+        """
+
+        # Validate detail level
+        DETAIL_LEVELS = ['1sec', '1min']
+        if detail_level not in DETAIL_LEVELS:
+            raise ValueError(f"detail_level {detail_level} should be one of {DETAIL_LEVELS}.")
+
+        # Validate date
+        start_date = check_date_format_wrapper(start_date)
+        end_date = check_date_format_wrapper(end_date)
+
+        # Validate both times or neither provided 

```

Comment:
This comment is a big vague, I misread it the first time.  Can you make it more clear by saying "Validate both start and end times, or whether neither time was provided" or something?

Reply:
done